Automattic/wp-calypso
# Stats: Add support for CSV download of UTM data

Related to #87687 

## Proposed Changes

Adds the "Download data as CSV" button to the UTM Summary page.

Differs from existing download buttons as they rely on the Redux setup with it's query component. Instead we use the data as fetched via our custom hook and generate CSV data based on that.

### Screenshot

<img width="770" alt="SCR-20240318-spzr" src="https://github.com/Automattic/wp-calypso/assets/40267301/f962833c-6247-456e-ae95-007eb4238970">

## Testing Instructions



* Launch the Live site.
* Visit the Traffic page on a site with UTM data.
* Enable the `stats/utm-module` flag.
* Scroll down to the UTM section and click "View all" or "View details".
* Or go directly to Summary page at: `/stats/day/utm/SITE_SLUG`
* Confirm the download button is enabled and works if there is UTM data to show.
* Click the download button and confirm the CSV data looks correct.
* Change the period of the request.
* Download again and confirm the CSV data reflects the updated data.
* Confirm the button is disabled if there are no results displayed.

Note: If you revert the last commit on the branch, you can test the module with Jetpack.com which has lots of UTM data to test with.

### Odd behaviour and a question

In my testing against Jetpack.com, I'm seeing odd results when requesting data via campaign. The API returns some unlabeled campaigns with children per this screenshot:

<img width="755" alt="SCR-20240318-synw" src="https://github.com/Automattic/wp-calypso/assets/40267301/a5242a10-5531-4dbf-bd88-a2e124a5a8f3">

I'm not sure how this is even possible but on the export side, this currently results in exports like the following:

- `"undefined > post title",COUNT_VALUE`

I looked at testing the campaign and returning an empty string but I'm not sure that is a better solution as it might not be immediately obvious that you have bad data. At least with the "undefined" string, it's easy to spot these rows and either delete or ignore as appropriate.

Any thoughts on if this should be handled differently, let me know!

## Pre-merge Checklist
